packages/plugin-betswirl/src/actions/roulette.ts

```diff
@@ -0,0 +1,347 @@
+    "numbers": [8, 11, 3, 9],
+    "token": "ETH",
+}
+\`\`\`
+
+** Message 3 **
+\`\`\`
+Spin a roulette for me
+\`\`\`
+
+** Response 3 **
+\`\`\`json
+{
+    "betAmount": null,
+    "numbers": null,
+    "token": null,
+}
+\`\`\`
+
+** Message 4 **
+\`\`\`
+I want to bet on the 8, 5, and 26
+\`\`\`
+
+** Response 4 **
+\`\`\`json
+{
+    "betAmount": null,
+    "numbers": [8, 5, 26],
+    "token": null,
+}
+\`\`\`
+
+** Message 5 **
+\`\`\`
+I want to bet 0.01 on 8, 5, and 26
+\`\`\`
+
+** Response 5 **
+\`\`\`json
+{
+    "betAmount": "0.01",
+    "numbers": [8, 5, 26],
+    "token": null,
+}
+\`\`\`
+
+** Message 6 **
+\`\`\`
+I want to bet my ETH on 8, 5, 26
+\`\`\`
+
+** Response 6 **
+\`\`\`json
+{
+    "betAmount": null,
+    "numbers": [8, 5, 26],
+    "token": "ETH",
+}
+\`\`\`json
+
+Here are the recent user messages for context:
+{{recentMessages}}
+`;
+export const rouletteAction: Action = {
+    name: "ROULETTE",
+    similes: [
+        "SPIN_A_ROULETTE",
+        "BETSWIRL_ROULETTE",
+        "BET_ON_ROULETTE",
+        "BET_ON_ROULETTE_BETSWIRL",
+    ],
+    description:
+        "Play the BetSwirl Roulette. The player is betting that the rolled number will be one of the chosen numbers.",
+    suppressInitialMessage: true,
+    validate: async (
+        _runtime: IAgentRuntime,
+        _message: Memory,
+        _state: State
+    ) => true,
+    handler: async (
+        runtime: IAgentRuntime,
+        message: Memory,
+        state: State,
+        _options: { [key: string]: unknown },
+        callback: HandlerCallback
+    ) => {
+        try {
+            elizaLogger.log("Starting ROULETTE handler...");
+
+            // Validate the chain
+            const wallet = state.moxieWalletClient as MoxieWalletClient;
+            const chainId = await getChainIdFromWallet();
+
+            // Initialize or update state
+            if (!state) {
+                state = (await runtime.composeState(message)) as State;
+            } else {
+                state = await runtime.updateRecentMessageState(state);
+            }
+            const context = composeContext({
+                state,
+                template: rouletteTemplate,
+            });
+            const rouletteDetails = await generateObject({
+                runtime,
+                context,
+                modelClass: ModelClass.SMALL,
+                schema: RouletteBetParameters,
+            });
+            const { numbers, betAmount, token } = rouletteDetails.object as {
+                numbers: Array<RouletteNumber>;
+                betAmount: string;
+                token: string;
+            };
+
+            // Validate face is heads or tails
+            if (!numbers || !numbers.length) {
+                throw new Error(
+                    `You must provide from 1 to ${MAX_SELECTABLE_ROULETTE_NUMBER} numbers between ${MIN_SELECTABLE_ROULETTE_NUMBER} and ${MAX_SELECTABLE_ROULETTE_NUMBER}. i.e. "Bet 0.07 ETH on 3, 18, 26, and 31"`
```

Comment:
Looks great

Can you also add brief explanations on what the number represents here? To inform the user what they're actually betting on when they're playing the roulette